module/apmprometheus/gatherer_test.go

```diff
@@ -134,6 +134,92 @@ func TestLabels(t *testing.T) {
 	}}, metrics)
 }
 
+func TestHistogram(t *testing.T) {
+	r := prometheus.NewRegistry()
+	h := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "histogram",
+			Help:    ".",
+			Buckets: []float64{1, 3, 5, 10, 15},
+		},
+		[]string{"code", "method"},
+	)
+	r.MustRegister(h)
+
+	h.WithLabelValues("200", "GET").Observe(3.4)
+	h.WithLabelValues("200", "GET").Observe(3.4)
+	h.WithLabelValues("200", "GET").Observe(3.4)
+
+	h.WithLabelValues("302", "GET").Observe(5.5)
+	h.WithLabelValues("302", "GET").Observe(5.5)
+	h.WithLabelValues("302", "GET").Observe(5.5)
+
+	h.WithLabelValues("302", "GET").Observe(11.2)
+	h.WithLabelValues("302", "GET").Observe(11.2)
+	h.WithLabelValues("302", "GET").Observe(11.2)
+
+	g := apmprometheus.Wrap(r)
+	metrics := gatherMetrics(g)[1:]
+
+	assert.Equal(t, []model.Metrics{{
+		Labels: model.StringMap{
+			{Key: "code", Value: "200"},
+			{Key: "method", Value: "GET"},
+		},
+		Samples: map[string]model.Metric{
+			"histogram": {
+				Type:   "histogram",
+				Values: []float64{4},
+				Counts: []uint64{3},
+			},
+		},
+	}, {
+		Labels: model.StringMap{
+			{Key: "code", Value: "302"},
+			{Key: "method", Value: "GET"},
+		},
+		Samples: map[string]model.Metric{
+			"histogram": {
+				Type:   "histogram",
+				Values: []float64{7.5, 10},
```

Comment:
I would expect this to be `7.5, 12.5`. I think there's an issue with the translation of the final bucket. We should only take the 2nd last bucket value when the final bucket's upper bound is +Inf, i.e. all the values greater than the greatest specified upper bound (15 in this case).

Reply:
Updated the code to reflect this, thanks for catching it!